src/Compilers/CSharp/Test/Semantic/Semantics/OverloadResolutionTests.cs

```diff
@@ -8000,13 +8019,13 @@ static void Main(string[] args)
 }
 ";
 
-            var compilation = CreateStandardCompilation(source1, options: TestOptions.DebugExe);
-
-            compilation.VerifyDiagnostics(
+            CreateStandardCompilation(source1, options: TestOptions.DebugExe, parseOptions: TestOptions.WithoutImprovedOverloadCandidates).VerifyDiagnostics(
     // (23,38): error CS0121: The call is ambiguous between the following methods or properties: 'Program.Bar<T, V>.Create(Func<T, bool>, params int[])' and 'Program.Bar<T, V>.Create(Func<T, V>, params int[])'
     //             var x = Bar<Goo, double>.Create(Goo.IsThing);
     Diagnostic(ErrorCode.ERR_AmbigCall, "Create").WithArguments("ConsoleApplication2.Program.Bar<T, V>.Create(System.Func<T, bool>, params int[])", "ConsoleApplication2.Program.Bar<T, V>.Create(System.Func<T, V>, params int[])").WithLocation(23, 38)
                 );
+            CreateStandardCompilation(source1, options: TestOptions.DebugExe).VerifyDiagnostics(
```

Comment:
This is surprising. Is it intended?
If yes, we should verify the semantic model (which overload was picked) and execution.